File: lib/py/uapi/internal/types/UHeaders.py
from typing import dict, TYPE_CHECKING

if TYPE_CHECKING:
    from uapi.internal.types.UFieldDeclaration import UFieldDeclaration


class UHeaders:
    def __init__(
            self,
            name: str,
            requestHeaders: dict[str, 'UFieldDeclaration'],
            responseHeaders: dict[str, 'UFieldDeclaration']
    ) -> None:
        self.name = name
        self.requestHeaders = requestHeaders
        self.responseHeaders = responseHeaders

File: lib/ts/uapi/internal/types/UHeaders.ts
import { UFieldDeclaration } from 'uapi/internal/types/UFieldDeclaration';

export class UHeaders {
    name: string;
    requestHeaders: { [key: string]: UFieldDeclaration };
    responseHeaders: { [key: string]: UFieldDeclaration };

    constructor(
        name: string,
        requestHeaders: { [key: string]: UFieldDeclaration },
        responseHeaders: { [key: string]: UFieldDeclaration },
    ) {
        this.name = name;
        this.requestHeaders = requestHeaders;
        this.responseHeaders = responseHeaders;
    }
}
